client/src/components/BlogPreview.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { Row, Col, Icon, Typography, Spin } from 'antd';
+import { Trans } from '@lingui/macro';
+import config from '../config';
+import BlogCard from './BlogCard';
+
+const { Title } = Typography;
+
+const { POSTS_URL } = config;
+
+const LIMIT = 4;
+
+export default ({ title, postSlug = null }) => {
+  const [state, setState] = useState({
+    posts: [],
+    requestError: false,
+    loading: true,
+  });
+
+  const loadPosts = useCallback(async (slug) => {
+    try {
+      const res = await fetch(`${POSTS_URL}/?limit=${LIMIT}&ordering=-published`);
+      if (res.ok) {
+        let { results: posts } = await res.json();
+        if (slug) {
+          posts = posts.filter((post) => post.slug !== slug);
+        }
+        if (posts.length > 3) {
+          posts.pop();
+        }
+
+        setState((prevState) => ({
+          ...prevState,
+          posts,
+          loading: false,
+          requestError: false,
+        }));
+      } else {
+        throw new Error(res.statusText);
+      }
+    } catch (err) {
+      setState((prevState) => ({
+        ...prevState,
+        posts: [],
+        requestError: true,
+        loading: false,
+      }));
+    }
+  }, []);
+
+  useEffect(() => {
+    loadPosts(postSlug);
```

Comment:
Why not just place the function inside the useEffect instead of creating a callback and use it inside the useEffect ?